src/lib.rs

```diff
@@ -25,6 +27,31 @@ impl Script {
 impl_from_core_type!(Script, BitcoinScriptBuf);
 impl_from_ffi_type!(Script, BitcoinScriptBuf);
 
+#[derive(Debug, Clone)]
+pub struct TxOut {
+    pub value: Arc<Amount>,
+    pub script_pubkey: Arc<Script>,
+}
+
+impl From<BitcoinTxOut> for TxOut {
+    fn from(tx_out: BitcoinTxOut) -> Self {
+        TxOut {
+            value: Arc::new(Amount(tx_out.value)),
+            script_pubkey: Arc::new(Script(tx_out.script_pubkey)),
+        }
+    }
+}
+
+impl From<TxOut> for BitcoinTxOut {
+    fn from(tx_out: TxOut) -> Self {
+        BitcoinTxOut {
+            value: tx_out.value.0,
+            script_pubkey: tx_out.script_pubkey.0.clone(),
```

Comment:
Curious if this needs to be cloned

Reply:
The clone is cheap with the `Arc`. Disregard

Reply:
Agreed

Reply:
thanks for that suggestion+feedback!

Reply:
We have not used this distinction (explicit costly vs cheap clones) in the past and I like it! Thanks for the great suggestions @tnull; will come in handy in a bunch of places.

@reez in this case we end up cloning the inner value anyway so increasing the count on the `tx_out.script_pubkey` is not required. You can revert and use the old `script_pubkey: tx_out.script_pubkey.0.clone(),` line here.

Reply:
Thanks tnull, this PR has been updated to use your suggestion here!

_the only small difference is I happened to be able to also use `arc.0` instead of `arc.0.clone()` for `Amount` since it implements the `Copy` trait from what I can see in [bitcoin_units](https://docs.rs/bitcoin-units/0.1.2/src/bitcoin_units/amount.rs.html#856)_